src/subscript/check_swatinit/check_swatinit.py

```diff
@@ -0,0 +1,612 @@
+    __SWL_TRUNC__,  # SWATINIT was less than SWL, water is gained.
+    __UNKNOWN__,  # If the situation is not understood by this script
+    __WATER__,  # All good, SWATINIT=1 below contact and SWAT=1
+]
+
+DESCRIPTION = "QC tool for SWATINIT vs SWAT in Eclipse runs"
+
+CATEGORY = "utility.eclipse"
+
+EXAMPLES = """
+.. code-block:: console
+
+  FORWARD_MODEL CHECK_SWATINIT(<DATAFILE>=<ECLBASE>, <OUTPUT>=check_swatinit.csv)
+
+where ``ECLBASE`` is already defined in your ERT config.
+"""
+
+
+def main():
+    """Executed when called from the command line.
+
+    Acts on command line arguments, loads data, performs qc and dumps to
+    CSV if requested."""
+    parser = get_parser()
+    args = parser.parse_args()
+
+    if args.DATAFILE.endswith(".csv"):
+        qc_frame = pd.read_csv(args.DATAFILE)
+    else:
+        eclfiles = ecl2df.EclFiles(args.DATAFILE)
+
+        # Fail hard if the deck is not suitable for this tool:
+        check_applicability(eclfiles)
+
+        qc_frame = make_qc_gridframe(eclfiles)
+
+        if args.output != "":
+            logger.info("Exporting CSV to %s", args.output)
+            qc_frame.to_csv(args.output, index=False)
+
+    if "SWATINIT" not in qc_frame:
+        print("Model did not use SWATINIT")
+        return
+    qc_vols = qc_volumes(qc_frame)
+    print(human_report_qc_vols(qc_vols))
+    qcsum = qc_vols["SWATINIT_WVOL"] + sum([qc_vols[qc_flag] for qc_flag in QC_FLAGS])
+    diff = qc_vols["SWAT_WVOL"] - qcsum
+    if not np.isclose(diff, 0, atol=1e-6):
+        print(f"Unexplained difference: {diff} Rm3")
+
+    print()
+    print(human_report_pc_scaling(qc_frame))
+
+    if args.eqlnum not in qc_frame["EQLNUM"].values:
+        sys.exit(f"Error: EQLNUM {args.eqlnum} does not exist in grid. No plotting.")
+    if args.plot or args.plotfile:
+        plotter.plot_qc_panels(qc_frame[qc_frame["EQLNUM"] == args.eqlnum])
+    if args.plot:
+        pyplot.show()
+    if args.plotfile:
+        print(f"Dumping plot to {args.plotfile}")
+        pyplot.savefig(args.plotfile)
+
+    if args.volplot or args.volplotfile:
+        plotter.wvol_waterfall(qc_vols)
+    if args.volplot:
+        pyplot.show()
+    if args.volplotfile:
+        print(f"Dumping volume plot to {args.volplotfile}")
+        pyplot.savefig(args.volplotfile)
+
+
+def check_applicability(eclfiles):
+    """Check that the input is relevant for usage with check_swatinit. This
+    function may raise exceptions, SystemExit or only give warnings"""
+
+    deck = eclfiles.get_ecldeck()
+
+    init = eclfiles.get_initfile()
+    if (
+        "SWATINIT" not in [initheader[0] for initheader in init.headers]
+        and "SWATINIT" not in deck
+    ):
+        logger.warning(
+            "INIT-file/deck does not have SWATINIT, this tool has limited use."
+        )
+
+    if "RPTRST" not in deck:
+        logger.warning(
+            "RPTRST not found in DATA-file, UNRST file is expected to be missing"
+        )
```

Comment:
Probably need to check that RPTRST is present in the SOLUTION section to ensure restart output at time 0?

Reply:
Do you know how to ask a `DeckKeyword` for which section it belongs to?

Reply:
No.. I guess just checking if report step 0 is present is easier.

Reply:
There is a `test_no_rptrst()` that checks that you get a `SystemExit` mentioning `UNRST` already if there is no data at report step 0. The code above is only meant as a hint.